tests/System.Text.Primitives.Tests/PrimitiveParserIntegerTests.cs

```diff
@@ -142,9 +142,9 @@ private byte[] UtfEncode(string s, bool utf16)
             new byte[] { 69 }, new byte[] { 101 },
         };
             
-        static EncodingData s_thaiEncoding = new EncodingData(s_thaiUtf8DigitsAndSymbols, TextEncoder.Utf8);
+        static TextEncoder s_thaiEncoder = TextEncoder.CreateUtf8Encoder(s_thaiUtf8DigitsAndSymbols);
 
-        static EncodingData s_utf8Encoding = new EncodingData(s_utf8digitsAndSymbols, TextEncoder.Utf8);
+        static TextEncoder s_utf8Encoder = TextEncoder.CreateUtf8Encoder(s_utf8digitsAndSymbols);
```

Comment:
Yes. We are aware of this. However, symbol encoding / decoding is just part of encoding process.

The other way seemed more confusing.